chains/solana/contracts/crates/common/src/lib.rs

```diff
@@ -0,0 +1,50 @@
+//! Contains common logic used across ccip programs. Common logic is versioned independently:
+//! v1 of the common logic may not necessarily be used by v1 of each particular program.
+use anchor_lang::prelude::*;
+
+pub mod seed;
+pub mod v1;
+
+#[error_code]
+pub enum CommonCcipError {
+    #[msg("The given sequence interval is invalid")]
+    // offset error code so that they don't clash with other programs
+    // (Anchor's base custom error code 6000 + offset 3000 = start at 9000)
+    InvalidSequenceInterval = 4000,
+    #[msg("Invalid pool accounts")]
+    InvalidInputsPoolAccounts,
+    #[msg("Invalid token accounts")]
+    InvalidInputsTokenAccounts,
+    #[msg("Invalid config account")]
+    InvalidInputsConfigAccounts,
+    #[msg("Invalid Token Admin Registry account")]
+    InvalidInputsTokenAdminRegistryAccounts,
+    #[msg("Invalid LookupTable account")]
+    InvalidInputsLookupTableAccounts,
+    #[msg("Invalid LookupTable account writable access")]
+    InvalidInputsLookupTableAccountWritable,
+}
+
+// Used only to enable the `account` attribute to enable account deserialization, for accounts
+// owned by the router (see TokenAdminRegistry below)
+declare_id!("Ccip842gzYHhvdDkSyi2YVCoAWPbYJoApMFzSxQroE9C");
```

Comment:
I think this is a problem, unless what tooling does accounts for this. We will use different addresses in different environments, so tooling has to replace the id where needed when they compile for each env. Anchor may use the value set here for ownership checks, so it is important that each time this is compiled, the right value is specified there.

It might just be the case that the way tooling is doing the compilation already matches this and there's no further work required, but please do check with them.

Reply:
Good shout, I asked that question [here](https://chainlink-core.slack.com/archives/C084CRQEQ5B/p1742547398330069). I do wonder if, aside from this, we can make their life easier with a little bit of `build.rs` magic (would be muc preferable to string substitution on the source if that's what's happening.)

Reply:
I had a sync conversation with tooling and my understanding after it is that this wouldn't be a problem: At worst there will be minor modifications needed on their end. 

Unrelated to this PR there may be opportunities for us to help alleviate some of the manual work when updating keys manually, but that's outside this scope.